### DISCUSSION TITLE
SHM/Sunnyportal findet keine Geräte aus dem SAE mehr / "sieht" den SAE nicht mehr

### DISCUSSION THREAD

**Discussion by Doenerteller** [camueller/SmartApplianceEnabler#518] [Installation und Betrieb]
Hallöchen,

mein SHM (v1.0)/Sunnyportal findet seit neuestem meine im SAE v.2.2.2 (frisch installiert) gepflegten Geräte nicht mehr.
Meine schon länger betriebenen Geräte/Verbraucher waren im Sunnyportal drin, haben aber seit neulich gar keine bzw. falsche Verbrauchswerte angezeigt. Daher hatte ich sie im Sunnyportal gelöscht um die Mülldaten los zu werden. Danach wollte ich sie einfach frisch wieder anlegen.

Nun werden sie leider nicht mehr gefunden, selbst wenn ich sie neu mit einer frischen Geräte-ID im SAE anlege.
/semp wird normal vom SAE bereitgestellt und sieht für mich erstmal OK aus. Die Geräte sind im /semp ON und ein Leistungsbezug 
wird übermittelt. Hier mal beispielhaft die Wärmepumpe

![image](https://user-images.githubusercontent.com/119921647/234560550-cd9a50a7-308b-481b-933d-7e7371053c98.png)

Im Sunnyportal ist bei mir auch die Option "Gelöschte Neugeräte wiederherstellen" komplett verschwunden. Früher™ konnte man sich von dort zuvor gelöschte Geräte wieder aus der Schublade holen, wenn man sie (mit der gleichen ID) wiederverwenden wollte.

Ich meine mich auch daran erinnern zu können, dass der SAE sich im Anlagenlogbuch im Sunnyportal immer mal wieder mit einer Meldung wie "Hey, habe einen EM Gerät mit SEMP gesehen" gezeigt hat. Habe ich das richtig in Erinnerung? Da sehe ich ihn nämlich auch nicht mehr.

Den Raspi mit SAE drauf und der SHM waren und sind auch weiterhin im gleichen LAN. Daran habe ich nichts verändert.
Ich kann den SHM vom SAE aus anpingen. Diverse Neustarts beider Gerätschaften wurden schon vollzogen.
Hat der SHM eigentlich eine Weboberfläche wo man sich einloggen kann um vielleicht mehr über seinen Zustand zu erfahren? 
Stumpf unter http://<SHM-IP>/ meldet er sich zumindest nicht.

Kann mir jemand Tips geben an welcher Stelle ich das debugging am sinnvollsten ansetzen sollte?
SAE log erscheint mir die falsche Stelle zu sein, da /semp augenscheinlich korret bereitgestellt wird.

Bin für jeden Tip dankbar!

Gruss

**Comment by Doenerteller** [comment]
Im Logbuch erscheint übrigens dashier

![image](https://user-images.githubusercontent.com/119921647/234573125-8a0935f4-30f9-420e-8008-722a305cdb63.png)

Das obere ist der EV Charger. Darunter dann noch ein Dingsbums. Soll das der SAE sein? Sieht mir nach einer abgeschnittenen Beschreibung aus. Steht da evtl. eigentlich die IP des SAE und die kommt einfach nicht komplett rüber zum HomeManager und daher
kann sich der HomeManager dann nicht mit dem SAE unterhalten?

Wie kann ich denn die "SEMP discovery" Daten einsehen, die der SAE in's LAN pustet? Vielleicht sind die unvollständig.

Ahh! Ich glaube der SAE verbreitet einen falsche IP!

![image](https://user-images.githubusercontent.com/119921647/234575476-8acb4174-71e1-49b3-a535-d1a0393cd2ef.png)


Ich meine mich zu erinner was in der Doku gelesen zu haben wie/wo man das einzustellen hat, wenn es auf anhieb nicht von selbst klappt. Die Suche beginnt...

**Comment by Doenerteller** [comment]
In den Java Optionen in /etc/default/smartapplianceenabler kann man das einstellen

![image](https://user-images.githubusercontent.com/119921647/234577640-a96d11c4-af02-4ce4-8361-17049e727b09.png)

Gemacht, neu gestartet, und jetzt sagt das Log die korrekte IP

![image](https://user-images.githubusercontent.com/119921647/234577739-4f1deeff-a0d4-4f51-9dbd-f9bbe3f3ae0d.png)


Mal abwarten was nun passiert.
Und warum das überhaupt so war...??? Vielleicht wegen meiner teils manuellen Installation. Ich habe schon Mosquitto und Pigpoid laufen und hatte mir daher das Installationsscript entsprechend angepasst vor demAausführen, dass er mir nicht in die bestehende Installation eingrätscht. Kann mich allerdings nicht daran erinnern irgendwas an IPs oder Java Optionen geschraubt zu haben.

**Comment by Doenerteller** [comment]
Nu guck mal einer an

![image](https://user-images.githubusercontent.com/119921647/234579813-37cce919-167a-4693-99ee-032d2a89ccea.png)

Haben wir nun also einen Fall von manueller Dösbaddlinstallation hier dokumentiert.
Schuss --> eigener Fuß.

**Reply by RobToas** [reply]
Hallo Doenerteller,
Ist es nicht verbunden mit meinem Thema?
<https://github.com/camueller/SmartApplianceEnabler/discussions/272>

VG,
Rob

**Comment by Doenerteller** [comment]
Nein, ich denke nicht. Bei mir war es eine falsche Konfiguration des SAE.
Allerdings kann ich aus Erfahrung sagen, dass das Sunnyportal seltsam (und langsam sowieso...) reagiert, wenn man irgendwelche Nummern wiederholt verwendet. Meiner Meinung nach hat das Ding irgendein Gedächtnis, welches sich merkwürdig verhält.
Daher vergebe ich IMMER neue Nummern, wenn ich ein Gerät nochmal neu anlegen muss/will.
In einem anderen Thread von mir hatte ich das beschrieben, glaube ich...frag mich jetzt bitte nicht, wie das Thema hieß. Wenn du ein Gerät löschst, dann ist das nicht weg. Das wandert in eine Liste von "zuvor gelöschter Geräte", die man auch wiederherstellen kann. Das Gerät ist daher nicht wirklich gelöscht und ich kann mir vorstellen, dass es daher bei erneuten Anlegen mit gleicher Nummer Probleme gibt.
